src/main.rs

```diff
@@ -111,20 +127,31 @@ fn block_level() -> impl Parser<NorgToken, Vec<NorgBlock>, Error = chumsky::erro
         .repeated()
         .at_least(1);
 
+    let extension_section = just(norg_char('('))
+        // .then(paragraph_segment.clone())
+        .ignore_then(
+            none_of([norg_char(')'), Newlines, SingleNewline, Eof])
+                .repeated()
+                .at_least(1),
```

Comment:
I'm personally against parsing strategies where we "parse until the end is found", especially since this greatly hinders the error generation capabilities (and also recovery strategy) - I think this should be changed to the following:
1. Parse the opening `(` char
2. Parse a single special char (any type, doesn't mater)
3. Parse whitespace
4. Parse the "parameters" bit until a `|` or `)` is found
5. Repeat from step 2 as many times as possible
6. Parse `)`

Then the actual grouping can be done in the third stage :)

Reply:
> Parse a single special char (any type, doesn't mater)

this seems to require making `x` a special character, right?

Reply:
On second look, this wasn't as bad as I thought it was. Changing the code to use `delimited_by` instead will help the error recovery. It still won't be as efficient as it could be if it was absolutely squeezed to the max, but this is more than good enough :)